### PR TITLE
Add golangci-lint to operator and executor test step and to core-builder image

### PR DESCRIPTION
**Pull request by ivan-valkov**
**What this PR does / why we need it**:

* Adds golangci-lint to the core-builder image
* Updates the version of core-builder used in executor and operator tests
* Run golangci-lint as part of the test make target in the executor and operator

The tests here will fail until we build and push the new image of the core-builder. Wanted to get your opinion on this change before building and pushing the new image.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

nope

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ivan-valkov**
You can assign the PR to them by writing `/assign @ivan-valkov` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by majolo — APPROVED** [review]
lgtm

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[majolo](https://github.com/SeldonIO/seldon-core/pull/3456#pullrequestreview-722488534 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [majolo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3456]: [Not Found]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3456]: [Not Found]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3456]: [Not Found]

**Pull request merged**
